Easier instructions and configuration for contributors

malwarefrank opened the pull request.

utkonos commented on setup.cfg:
```diff
@@ -0,0 +1,5 @@
+max-line-length=120
+
+[pydocstyle]
+add-ignore=D101,D102,D104,D203,D205,D208,D209,D213,D300,D400,D401,D403,D406,D407,D413,D415
```
The other changes are fine. This one needs work. The error codes are very very specific because PLY has some idiosyncratic syntax that is required and not Python standard. If the codes in the configuration you propose can be applied in the exact same sets to the exact same files, this can be merged. If not, this can't be merged.

utkonos commented:
Sorry about those repeated comments. I absolutely hate the UI for reviews.

malwarefrank commented:
I did not readily an easy way to make some exceptions apply to some files and some to others, so I reverted and just moved the max line setting into setup.cfg

utkonos commented:
@malwarefrank I know that the complexity of unit testing and dev environment setup was a pain for you according to this PR. I am currently working on a rewrite of plyara from the ground up which will retain the same API and CLI from 2.x with some major enhancements. I kept your pain point in mind while working on 3.0.0. I have simplified the whole entire test suite, dev environment, and CI. Please take a look at the setup in the following PR and let me know if this works better than how it was in 2.x:
https://github.com/plyara/plyara/tree/prep-3.0

malwarefrank commented:
Yes, it looks much easier.  I have a few small changes.  Will submit a PR to that branch
